steel/src/history/beacon_roots.rs

```diff
@@ -14,7 +14,24 @@
 
 use crate::{MerkleTrie, StateAccount};
 use alloy_primitives::{address, b256, keccak256, uint, Address, B256, U256};
+use revm::{
+    primitives::{AccountInfo, Bytecode},
+    Database,
+};
 use serde::{Deserialize, Serialize};
+use std::convert::Infallible;
+
+/// The length of the buffer that stores historical entries, i.e., the number of stored
+/// timestamps and roots.
+pub const HISTORY_BUFFER_LENGTH: U256 = uint!(8191_U256);
+/// Address where the contract is deployed.
+pub const ADDRESS: Address = address!("000F3df6D732807Ef1319fB7B8bB8522d0Beac02");
+
+/// Hash of the contract's address, where the contract is deployed.
+const ADDRESS_HASH: B256 =
+    b256!("37d65eaa92c6bc4c13a5ec45527f0c18ea8932588728769ec7aecfe6d9f32e42");
+/// Hash of the deployed EVM bytecode.
+const CODE_HASH: B256 = b256!("f57acd40259872606d76197ef052f3d35588dadf919ee1f0e3cb9b62d3f4b02c");
 
 /// Enum representing possible errors that can occur within the `BeaconRootsContract`.
 #[derive(Debug, thiserror::Error)]
```

Comment:
We might add `non_exhaustive` to allow us to add more options in the future without breaking the API.
```suggestion
#[derive(Debug, thiserror::Error)]
#[non_exhaustive]
```